wasm/jit/engine.go

```diff
@@ -408,9 +339,44 @@ func (e *engine) maybeGrowStack(maxStackPointer uint64) {
 	// TODO: maybe better think about how to shrink the stack as well.
 }
 
-func (e *engine) exec(f *compiledWasmFunction) {
+func (e *engine) execHostFunction(f *compiledFunction, ctx *wasm.HostFunctionCallContext) {
```

Comment:
this needs a doc of what's going on or comments about the conversion of in/out inside the function or both because looking at this is intimidating otherwise 🏃